Lib/Coproduct.hpp

```diff
@@ -275,8 +275,8 @@ class Coproduct<A, As...>
   using Self = Coproduct<A, As...>;
   using Content = decltype(_content);
 
-  /** unsafe default constructor. tag as well as content will be uninit */
-  Coproduct() {}
+  /** unsafe default constructor, content will be uninit */
+  Coproduct() : _tag(std::numeric_limits<unsigned>::max()) {}
```

Comment:
Try and make this as likely as possible to blow up if we don't initialise it correctly.